Libplanet/Net/Messages/Message.cs

```diff
@@ -341,14 +351,30 @@ private enum MessageFrame
 
             var message = new NetMQMessage();
 
+            // Make signature
             // Write body (by concrete class)
             foreach (NetMQFrame frame in DataFrames)
             {
                 message.Append(frame);
             }
 
             // Write headers. (inverse order)
-            message.Push(key.Sign(message.ToByteArray()));
+            message.Push(timestamp.Ticks);
+            message.Push(SerializePeer(peer));
+            message.Push((byte)Type);
+            message.Push(version.Token);
+
+            var signature = key.Sign(message.ToByteArray());
+            message.Clear();
+
+            // Make message
+            foreach (NetMQFrame frame in DataFrames)
+            {
+                message.Append(frame);
+            }
+
+            // Write headers. (inverse order)
+            message.Push(signature);
```

Comment:
According to [this](https://github.com/zeromq/netmq/blob/0b98a648ef0dd7933b5819257dee97e64bfb6420/src/NetMQ/NetMQMessage.cs#L48), we can make `NetMQMessage` with an `IEnumerable` instance. How about use `IList` typed message first before signing in order to inject signature easily without make message instance twice?

Reply:
According to [this](https://github.com/zeromq/netmq/blob/0b98a648ef0dd7933b5819257dee97e64bfb6420/src/NetMQ/NetMQMessage.cs#L249), there is a problem that the implementation of the part where the type conversion occurs when pushing is partly imported into `ToMQMessage`. So, what do you think we're trying to change in the following way? 

```suggestion
            byte[] signature = key.Sign(message.ToByteArray());
            List<NetMQFrame> frames = message.ToList();
            frames.Insert((int)MessageFrame.Sign, new NetMQFrame(signature));
            message = new NetMQMessage(frames);
```